### ISSUE TITLE
[Feature request] Check Alarm MetricDataQuery Id format

### ISSUE THREAD

**Issue by Graham42** [aws-cloudformation/cfn-python-lint#1944]
*cfn-lint version: 0.47.2* 

When creating an AWS::CloudWatch::Alarm and [using metric math](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/using-metric-math.html) the `Id` field of each [`MetricDataQuery`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudwatch-alarm-metricdataquery.html) must start with a lowercase letter. Without this, CloudFormation reports the not-so-helpful error `Invalid metrics list`. It would be awesome to have a rule to catch this so we're not left in the dark guess what's wrong with the metric list definition.

<details>
<summary>Sample template:</summary>

```yaml
Resources:
  ApiGateway5xxAlarm:
    Type: AWS::CloudWatch::Alarm
    Properties:
      AlarmDescription: Api Gateway Errors
      AlarmName: ApiGatewayAlarm
      ComparisonOperator: GreaterThanUpperThreshold
      EvaluationPeriods: 1
      Metrics:
      - Expression: ANOMALY_DETECTION_BAND(5xxErrors, 2)
        Id: ad1
      - Id: 5xxErrors # <<< this Id should start with a lowercase letter
        MetricStat:
          Metric:
            MetricName: 5XXError
            Namespace: AWS/ApiGateway
          Period: !!int 86400
          Stat: Sum
      ThresholdMetricId: ad1
      TreatMissingData: ignore
```

</details>

Links to the (official) documentation:

  - [AWS::CloudWatch::Alarm MetricDataQuery](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudwatch-alarm-metricdataquery.html)
  - [Using metric math](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/using-metric-math.html)

* [x] Validate if the issue still exists with the latest version of `cfn-lint` and/or the latest Spec files
